pallets/delegation/src/lib.rs

```diff
@@ -507,22 +507,24 @@ impl<T: Config> Pallet<T> {
 
 	// Creates a new root node with the given details and store the new hierarchy in
 	// the hierarchies storage and the new root node in the nodes storage.
-	fn create_and_store_new_hierarchy(
+	pub(crate) fn create_and_store_new_hierarchy(
 		root_id: DelegationNodeIdOf<T>,
 		hierarchy_details: DelegationHierarchyDetails<T>,
 		hierarchy_owner: DelegatorIdOf<T>,
-	) {
+	) -> DelegationNode<T> {
 		let root_node = DelegationNode::new_root_node(root_id, DelegationDetails::default_with_owner(hierarchy_owner));
-		<DelegationNodes<T>>::insert(root_id, root_node);
+		<DelegationNodes<T>>::insert(root_id, root_node.clone());
```

Comment:
Just noticing them in random places to remind to update the syntax.

```suggestion
		DelegationNodes::<T>::insert(root_id, root_node.clone());
```